libsolidity/analysis/IntegerOverflowBounds.cpp

```diff
@@ -0,0 +1,267 @@
+#include <libsolidity/analysis/IntegerOverflowBounds.h>
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+namespace dev
+{
+namespace solidity
+{
+
+void IntegerOverflowBounds::performBinaryOp(
+		Token::Value const& op,
+		OverflowBounds const& left,
+		OverflowBounds const& right
+		)
+{
+	auto const lbounds = dynamic_cast<IntegerOverflowBounds const*>(&left);
+	auto const rbounds = dynamic_cast<IntegerOverflowBounds const*>(&right);
+	solAssert((lbounds && rbounds), "Passed non-integer bounds objects to binary op for an integer bounds");
+	performBinaryOp(op, *lbounds, *rbounds);
+}
+	void IntegerOverflowBounds::performBinaryOp(
```

Comment:
I somehow expected this function to be static and returning a new `IntegerOverflowBounds`. Why is that not the case?